src/lib/types/Tool.ts

```diff
@@ -0,0 +1,51 @@
+type ToolInput =
+	| {
+			description: string;
+			type: string;
+			required: true;
+	  }
+	| {
+			description: string;
+			type: string;
+			required: false;
+			default: string | number | boolean;
+	  };
+
+export interface Tool {
+	name: string;
+	displayName?: string;
+	description: string;
+	parameterDefinitions: Record<string, ToolInput>;
+	spec?: string;
+	isOnByDefault?: true; // will it be toggled if the user hasn't tweaked it in settings ?
+	isLocked?: true; // can the user enable/disable it ?
+	isHidden?: true; // should it be hidden from the user ?
+}
+
+export type ToolFront = Pick<
+	Tool,
+	"name" | "displayName" | "description" | "isOnByDefault" | "isLocked"
+>;
+
+export enum ToolResultStatus {
+	Success = "success",
+	Error = "error",
+}
+interface ToolResultSuccess {
```

Comment:
same comment as https://github.com/huggingface/chat-ui/pull/996#discussion_r1606499556 for `ToolResultSuccess, ToolResultError` to create `BaseToolResult`